File: Python/big-o-rule-2/main.py
from profile import run as performance_run

BOXES = [1, 2, 3, 4, 5]


def log_first_two_boxes(boxes):
    """
    Prints the first two boxes in the given list.
    Time complexity: O(1) -> Constant Time

    Parameters:
    boxes (list): The list of boxes.

    Returns:
    None
    """
    print(boxes[0])  # O(1)
    print(boxes[1])  # O(1)


performance_run('log_first_two_boxes(BOXES)')

File: Go/big-o-rule-2/main.go
package main

import "fmt"

var boxes = [5]int32{1, 2, 3, 4, 5}

func main() {
	logFirstTwoBoxes(boxes)
}

/*
logFirstTwoBoxes prints the first two elements of the given array of boxes.
Time complexity: O(1) + O(1) = O(2) => O(1)

Parameters:

	*- boxes: array of int32 elements

Return:

	*- void
*/
func logFirstTwoBoxes(boxes [5]int32) {
	fmt.Println(boxes[0]) // O(1)
	fmt.Println(boxes[1]) // O(1)
}
